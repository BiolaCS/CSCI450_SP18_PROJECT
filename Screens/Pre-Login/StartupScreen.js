import React from 'react';
import {StyleSheet, Text, View, Image, Modal, TextInput, Alert, Animated, BackHandler, Keyboard} from 'react-native';
import * as firebase from 'firebase';
import { NavigationActions } from 'react-navigation';

import RoundedButton from '../../Components/RoundedButton';

const firebaseConfig = {
  apiKey: "AIzaSyAJXp7SBUPGRTPo-5qYM-T78mP8DEuBsog",
  authDomain: "commune-265d9.firebaseapp.com",
  databaseURL: "https://commune-265d9.firebaseio.com",
  projectId: "commune-265d9",
  storageBucket: "commune-265d9.appspot.com",
  messagingSenderId: "697540841037"
};

export default class StartupScreen extends React.Component {
  constructor() {
    super();
    console.ignoredYellowBox = [
      'Setting a timer'
    ];
    if (!firebase.apps.length) { // Prevent more than one instance
      firebase.initializeApp(firebaseConfig);
    }



    // Passed into the firebase auth system
    // For now all we need to worry about is if the user is signed in
    this.state = {
      loading: true,
      userEmail: '',
      userPassword: '',
      userHasTakenQuiz: false,
      offsetY: new Animated.Value(0),
    };
  }

  componentDidMount() {
    this.authSubscription = firebase.auth().onAuthStateChanged((user) => {
      this.setState({
        loading: false,
        user,
      });

      if (this.state.user) { // User has been logged in
        // Check if the user has anything data in firebase
        var userId = firebase.auth().currentUser.uid;
        this.checkQuizStatus(userId);
      }
    });
    this.keyboardDidHideListener = Keyboard.addListener('keyboardDidHide', this.changeTextBoxPositionsOriginal.bind(this));
  }

  componentWillUnmount() {
    this.authSubscription();
    this.keyboardDidHideListener.remove();
  }

  checkQuizStatus(userId) {
    firebase.database().ref('/users/' + userId).once('value').then((snapshot) => {

      if (snapshot.val() !== null) { // User has info already
        if (snapshot.val().hasTakenQuiz === true) { // They have taken the quiz
          // Props are inaccessible at this point for some reason
          this.navigateHome();
        }
        else { // They have not taken the quiz
          this.navigateQuiz();
        }
      }
      else { // User is brand new
        // Set quiz flag
        firebase.database().ref('users/' + userId).set({
          hasTakenQuiz: false
        });
        this.navigateQuiz();
      }
    })
    .catch(function(error) {
      console.log(error);
    });
  }

  navigateHome() {
    // Object to reset the navigation stack
    const resetAction = NavigationActions.reset({
      index: 0,
      actions: [NavigationActions.navigate({ routeName: 'Home' })],
    });

    this.props.navigation.dispatch(resetAction);
  }

  navigateQuiz() {
    // Alert the user that they have to take the quiz
    Alert.alert(
      "First time user detected!",
      "Please take the personality quiz",
      [
        {text: 'OK'},
      ],
      { cancelable: false }
    )
    // Object to reset the navigation stack
    const resetAction = NavigationActions.reset({
      index: 0,
      actions: [NavigationActions.navigate({ routeName: 'Questionnaire' })],
    });

    this.props.navigation.dispatch(resetAction);
  }

  loginUser() {
    console.log("logging in user");
    firebase.auth().signInWithEmailAndPassword(this.state.userEmail, this.state.userPassword).catch(function(error) {
      Alert.alert(
        error.code,
        error.message,
        [
          {text: 'OK'},
        ],
        { cancelable: false }
      )
    });
  }

  signupUser() {
    console.log("signing up user");
    firebase.auth().createUserWithEmailAndPassword(this.state.userEmail, this.state.userPassword).catch(function(error) {
        // Handle Errors here.
      Alert.alert(
        error.code,
        error.message,
        [
          {text: 'OK'},
        ],
        { cancelable: false }
      )
    });
  }

  logoutUser() {
    console.log("signing out");
    firebase.auth().signOut().then(function() {
      console.log("signed out");
    }).catch((error) => {
      console.log(error);
    })
  }

  changeTextBoxPositions() {
    console.log("moving text box");
    Animated.timing(this.state.offsetY,
      {toValue: -150}
    ).start();
  }

  changeTextBoxPositionsOriginal() {
    console.log("moving text box to original spot");
    Animated.timing(this.state.offsetY,
      {toValue: 0}
    ).start();
  }

  render() {

    return (
        <View style={styles.container}>



            <Animated.View style={{transform : [{translateY: this.state.offsetY}],
                                  flex: 1,
                                  alignItems: 'center',
                                  justifyContent: 'center',
                                  height: 100}}>

                <Image source={require('../../Images/Logo.png')} style={styles.logoNoKeyboard}/>

                <TextInput
                    onChangeText={(text) => {this.setState({userEmail: text}); }}
                    onFocus={this.changeTextBoxPositions.bind(this)}
                    onSubmitEditing={this.changeTextBoxPositionsOriginal.bind(this)}
                    placeholder={'Email'}
                    placeholderTextColor= '#000000'
                    style={styles.textInput}
                />

                <TextInput
                    onChangeText={(text) => {this.setState({userPassword: text}); }}
                    onFocus={this.changeTextBoxPositions.bind(this)}
                    onSubmitEditing={this.changeTextBoxPositionsOriginal.bind(this)}
                    placeholder={'Password'}
                    placeholderTextColor= '#000000'
                    secureTextEntry = {true}
                    style={styles.textInput}
                />

            </Animated.View>

            <RoundedButton onPress={() => {this.loginUser();}}>
                  Login
            </RoundedButton>

            <RoundedButton onPress={() => {this.signupUser();}}>
                  Sign-Up
            </RoundedButton>

            <RoundedButton onPress={() => {this.logoutUser();}}>
                  Logout
            </RoundedButton>

      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#d6edf5',
    alignItems: 'center',
    justifyContent: 'center',
  },
  textInput: {
    borderRadius: 5,
    color: '#000000',
    width: 250,
    height: 40,
    backgroundColor: '#f0f0f0',
    margin: 10,
    marginRight: 25,
    marginLeft: 25,
  },
  logoNoKeyboard: {
    justifyContent: 'center',
    width: 150,
    height: 193,
    marginBottom: 30,
    margin: 10,
    marginRight: 25,
    marginLeft: 25,
    borderRadius: 5,
  }
});
