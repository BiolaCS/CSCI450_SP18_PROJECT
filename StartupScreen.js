import React from 'react';
import {StyleSheet, Text, View, Image, Modal, TextInput, Alert } from 'react-native';
import * as firebase from 'firebase';

import RoundedButton from './App/Components/RoundedButton';
import FullButton from './App/Components/FullButton';

const firebaseConfig = {
  apiKey: "AIzaSyAJXp7SBUPGRTPo-5qYM-T78mP8DEuBsog",
  authDomain: "commune-265d9.firebaseapp.com",
  databaseURL: "https://commune-265d9.firebaseio.com",
  projectId: "commune-265d9",
  storageBucket: "commune-265d9.appspot.com",
  messagingSenderId: "697540841037"
};

export default class StartupScreen extends React.Component {
  constructor() {
    super();
    if (!firebase.apps.length) { // Prevent more than one instance
      firebase.initializeApp(firebaseConfig);
    }
    // Passed into the firebase auth system
    // For now all we need to worry about is if the user is signed in
    this.state = {
      loading: true,
      userEmail: '',
      userPassword: '',
    };
  }

  componentDidMount() {
    this.authSubscription = firebase.auth().onAuthStateChanged((user) => {
      this.setState({
        loading: false,
        user,
      });
      console.log(user);
      if (this.state.user) { // User has been logged in
        this.props.navigation.navigate('Login', {});
      }
    });
  }

  componentWillUnmount() {
    this.authSubscription();
  }

  loginUser() {
    console.log("logging in user");
    firebase.auth().signInWithEmailAndPassword(this.state.userEmail, this.state.userPassword).catch(function(error) {
      Alert.alert(
        error.code,
        error.message,
        [
          {text: 'OK'},
        ],
        { cancelable: false }
      )
    });
  }

  signupUser() {
    console.log("signing up user");
    firebase.auth().createUserWithEmailAndPassword(this.state.userEmail, this.state.userPassword).catch(function(error) {
        // Handle Errors here.
      Alert.alert(
        error.code,
        error.message,
        [
          {text: 'OK'},
        ],
        { cancelable: false }
      )
    });
  }

  logoutUser() {
    console.log("signing out");
    firebase.auth().signOut().then(function() {
      console.log("signed out");
    }).catch((error) => {
      console.log(error);
    })
  }

  render() {
    var {navigate} = this.props.navigation;

    return (
      <View style={styles.container}>

        <Image source={require('./App/Images/Logo.png')} style={styles.logoNoKeyboard}/>

        <TextInput
          style={styles.textInput}
          placeholder={'Username'}
          placeholderTextColor= '#000000'
          onChangeText={(text) => {this.setState({userEmail: text}); }}
        />

        <TextInput
          style={styles.textInput}
          placeholder={'Password'}
          placeholderTextColor= '#000000'
          secureTextEntry = {true}
          onChangeText={(text) => {this.setState({userPassword: text}); }}
        />

      <RoundedButton onPress={() => {this.loginUser(navigate);}}>
            Login
      </RoundedButton>

      <RoundedButton onPress={() => {this.signupUser();}}>
            Sign-Up
      </RoundedButton>

      <RoundedButton onPress={() => {this.logoutUser();}}>
            Logout
      </RoundedButton>

      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#84C9E0',
  alignItems: 'center',
    justifyContent: 'center',
  },
  textInput: {
    borderRadius: 5,
    color: '#000000',
    width: 250,
    height: 40,
    backgroundColor: '#f0f0f0',
    margin: 10,
    marginRight: 25,
    marginLeft: 25,
  },
  logoNoKeyboard: {
    width:150,
    height: 193,
    marginBottom: 60,
  }
});
