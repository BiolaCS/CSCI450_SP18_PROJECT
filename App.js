import React from 'react'
import {StyleSheet, Text, View, Modal } from 'react-native'
import { StackNavigator } from 'react-navigation';
import StartupScreen from './StartupScreen'
import LoginScreen from './LoginScreen'
import SignupScreen from './SignupScreen'
import QuestionnaireScreen from './QuestionnaireScreen'
import PostJoinScreen from './PostJoinScreen'


const Navigation = StackNavigator({
  Home: { screen: StartupScreen }, 
  Login: { screen: LoginScreen },
  Signup: { screen: SignupScreen },
  Questionnaire: { screen: QuestionnaireScreen },
  PostJoin: { screen: PostJoinScreen },
});

export default Navigation;
