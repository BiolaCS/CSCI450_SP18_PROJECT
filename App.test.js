import React from 'react';
import App from './App';
import StartupScreen from './Screens/Pre-Login/StartupScreen';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<StartupScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toBeTruthy();
});
