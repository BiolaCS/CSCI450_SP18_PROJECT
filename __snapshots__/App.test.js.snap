// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#d6edf5",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    collapsable={undefined}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 100,
        "justifyContent": "center",
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 5,
          "height": 193,
          "justifyContent": "center",
          "margin": 10,
          "marginBottom": 30,
          "marginLeft": 25,
          "marginRight": 25,
          "width": 150,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Email"
      placeholderTextColor="#000000"
      style={
        Object {
          "backgroundColor": "#f0f0f0",
          "borderRadius": 5,
          "color": "#000000",
          "height": 40,
          "margin": 10,
          "marginLeft": 25,
          "marginRight": 25,
          "width": 250,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Password"
      placeholderTextColor="#000000"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#f0f0f0",
          "borderRadius": 5,
          "color": "#000000",
          "height": 40,
          "margin": 10,
          "marginLeft": 25,
          "marginRight": 25,
          "width": 250,
        }
      }
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#e73536",
        "borderRadius": 5,
        "height": 45,
        "justifyContent": "center",
        "marginHorizontal": 25,
        "marginVertical": 10,
        "opacity": 1,
        "width": 250,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginVertical": 10,
          "textAlign": "center",
        }
      }
    >
      LOGIN
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#e73536",
        "borderRadius": 5,
        "height": 45,
        "justifyContent": "center",
        "marginHorizontal": 25,
        "marginVertical": 10,
        "opacity": 1,
        "width": 250,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginVertical": 10,
          "textAlign": "center",
        }
      }
    >
      SIGN-UP
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#e73536",
        "borderRadius": 5,
        "height": 45,
        "justifyContent": "center",
        "marginHorizontal": 25,
        "marginVertical": 10,
        "opacity": 1,
        "width": 250,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginVertical": 10,
          "textAlign": "center",
        }
      }
    >
      LOGOUT
    </Text>
  </View>
</View>
`;
